openpype/hosts/houdini/plugins/publish/collect_cache_farm.py

```diff
@@ -0,0 +1,87 @@
+import os
+import pyblish.api
+import hou
+from openpype.hosts.houdini.api import lib, colorspace
+from openpype.hosts.houdini.api.lib import get_color_management_preferences
+
+
+class CollectDataforCache(pyblish.api.InstancePlugin):
+    """Collect data for caching to Deadline."""
+
+    order = pyblish.api.CollectorOrder + 0.04
+    families = ["ass", "pointcache",
+                "mantraifd", "redshiftproxy"]
+    hosts = ["houdini"]
+    targets = ["local", "remote"]
+    label = "Collect Data for Cache"
+    chunkSize = 999999
+
+    def process(self, instance):
+        if instance.data.get("farm", True):
+            for family in instance.data["families"]:
+                instance.data["family"] = family
+                if family == "bgeo" or "abc":
+                    instance.data["family"] = "pointcache"
+                instance.data["subset"] = instance.name
+            # instance.data["expected_families"]
+            # only created for farm caching
+            instance.data["expected_families"] = list()
```

Comment:
Why is this `expected_families` with an underscore even though we also use `expectedFiles` just below?

Reply:
`expected_families `is not the same with the `expectedFile`. `expected_families `is to make sure the file transmission of submit job would be in the correct family such as `pointcache` and `ifd`

Reply:
They are two different variables, that's fine. I'm just saying why does the formatting differ? Why is one lowercase with an underscore and the other _lowerCamelCase_.

Reply:
By the way, the concept of defining in the data what the final published family (let's refer to it as the product-type) should be instead of inheriting all of the families during the Pyblish publish process is something that was done before here: https://github.com/ynput/OpenPype/blob/c10781a662e9dd85839d8eb786e471b0709cbccf/openpype/hosts/houdini/plugins/publish/collect_usd_bootstrap.py#L85 and https://github.com/ynput/OpenPype/blob/4c280328b85b8cccbe1f76bf8ffcfc26ae34df94/openpype/hosts/houdini/plugins/publish/collect_instances_usd_layered.py#L83 where it was named `publishFamilies`. It hasn't gotten much use in OpenPype and I have no idea if that logic actually works - but just so you know there is basically similar functionality as your `expectedFamilies` where it was named `publishFamilies` instead.

Reply:
> By the way, the concept of defining in the data what the final published family (let's refer to it as the product-type) should be instead of inheriting all of the families during the Pyblish publish process is something that was done before here:
> 
> https://github.com/ynput/OpenPype/blob/c10781a662e9dd85839d8eb786e471b0709cbccf/openpype/hosts/houdini/plugins/publish/collect_usd_bootstrap.py#L85
> 
> and
> https://github.com/ynput/OpenPype/blob/4c280328b85b8cccbe1f76bf8ffcfc26ae34df94/openpype/hosts/houdini/plugins/publish/collect_instances_usd_layered.py#L83
> 
> where it was named `publishFamilies`. It hasn't gotten much use in OpenPype and I have no idea if that logic actually works - but just so you know there is basically similar functionality as your `expectedFamilies` where it was named `publishFamilies` instead.

Will rename it to `publishFamilies` in the next PR